src/elfpy/pricing_models/base.py

```diff
@@ -39,71 +34,45 @@ def calc_out_given_in(
         """Calculate fees and asset quantity adjustments"""
         raise NotImplementedError
 
-    # TODO: Use the MarketState class.
     @abstractmethod
     def calc_lp_out_given_tokens_in(
         self,
         d_base: float,
-        share_reserves: float,
-        bond_reserves: float,
-        base_buffer: float,
-        init_share_price: float,
-        share_price: float,
-        lp_reserves: float,
         rate: float,
-        time_remaining: float,
-        stretched_time_remaining: float,
+        market_state: MarketState,
+        time_remaining: StretchedTime,
     ) -> tuple[float, float, float]:
         """Computes the amount of LP tokens to be minted for a given amount of base asset"""
-        # TODO: This pylint will be resolved when we switch to MarketState
-        # pylint: disable=too-many-arguments
         raise NotImplementedError
 
-    # TODO: Use the MarketState class.
     @abstractmethod
     def calc_lp_in_given_tokens_out(
         self,
         d_base: float,
-        share_reserves: float,
-        bond_reserves: float,
-        base_buffer: float,
-        init_share_price: float,
-        share_price: float,
-        lp_reserves: float,
         rate: float,
-        time_remaining: float,
-        stretched_time_remaining: float,
+        market_state: MarketState,
+        time_remaining: StretchedTime,
     ) -> tuple[float, float, float]:
         """Computes the amount of LP tokens to be minted for a given amount of base asset"""
-        # TODO: This pylint will be resolved when we switch to MarketState
-        # pylint: disable=too-many-arguments
         raise NotImplementedError
 
-    # TODO: Use the MarketState class.
     @abstractmethod
     def calc_tokens_out_given_lp_in(
         self,
         lp_in: float,
-        share_reserves: float,
-        bond_reserves: float,
-        base_buffer: float,
-        init_share_price: float,
-        share_price: float,
-        lp_reserves: float,
         rate: float,
-        time_remaining: float,
-        stretched_time_remaining: float,
+        market_state: MarketState,
+        time_remaining: StretchedTime,
     ) -> tuple[float, float, float]:
         """Calculate how many tokens should be returned for a given lp addition"""
-        # TODO: This pylint will be resolved when we switch to MarketState
-        # pylint: disable=too-many-arguments
         raise NotImplementedError
 
     @abstractmethod
     def model_name(self) -> str:
         """Unique name given to the model, can be based on member variable states"""
         raise NotImplementedError
 
+    # TODO: FIXME: This does not work with yieldspace/hyperdrive
```

Comment:
How so? It looks fine to me, so I'd be curious to hear what isn't working.

Reply:
I removed the TODO, because I'm not certain now.